### DISCUSSION TITLE
Pauta colaborativa para as Reuniões de Mantenedoras do Querido Diário

### DISCUSSION THREAD

**Discussion by trevineju** [okfn-brasil/querido-diario-comunidade#10] [General]
Olá, comunidade!

Na última reunião mensal de pessoas mantenedoras do Querido Diário, recebemos uma sugestão para adotarmos um esquema de pauta colaborativa, ou seja, abrir uma via em que vocês possam sinalizar pontos de pauta que gostariam de incluir na reunião. Nós gostamos muito da ideia e pretendemos adotá-la. Pensamos no seguinte esquema para essa interação:

1. Todas as pessoas podem responder esta thread com sugestões do que gostaria de incluir;
2. Outras pessoas podem votar, reagindo com um 👍, nas sugestões que considerarem mais importantes para usarmos como métrica de interesse e privilegiarmos as mais votadas; 
3. Buscaremos incluir as de maior interesse, porém pode acontecer de precisarmos priorizar outras excepcionalmente. Nesse caso, nós incluiremos a sugestão da pauta aqui na thread sinalizando que ela tem urgência.
4. As sugestões serão acumulativas. As que não entrarem na pauta do mês, ficarão em espera para a próxima oportunidade. 

As reuniões, vale lembrar, não são de pauta única. Temos a expectativa de, mesmo quando precisarmos pautar algo prioritariamente (conforme o item 3), poderemos seguir dando vazão às sugestões da comunidade aos poucos.

Assim, damos o pontapé para que vocês sugiram o que gostariam de incluir em pauta a partir de agora. Sintam-se muito livres a propor pontos que perceberem precisar de atenção.

**Comment by ogecece** [comment]
Adicionar mais tipos de metadados de municípios, possibilitando mais filtros de busca e análise mais rica de tendências para gestoras públicas, pesquisadoras, etc.

#9

**Reply by trevineju** [reply]
Entrou na pauta da reunião 13, de Abril/2022: https://hackmd.io/fJ-Zsz3IRbSeKtPktJYNSA. 
Sugestões de outros metadados podem ser incluídos na discussion #9

**Comment by ogecece** [comment]
Retomar a discussão sobre okfn-brasil/querido-diario#546 para chegarmos num formato satisfatório

**Comment by ogecece** [comment]
Retomar a discussão sobre como aproveitar cidades que não publicam o diário inteiro (páginas, atos, etc).

**Comment by trevineju** [comment]
Acordos para o repositório `querido-diário`: 
- Apresentar quadro: https://github.com/orgs/okfn-brasil/projects/12
- Discussões:
-- Onde centralizar informações de mapeamento para sistemas replicáveis? (iniciados em [querido-diario#724](https://github.com/okfn-brasil/querido-diario/issues/724), [querido-diario#752](https://github.com/okfn-brasil/querido-diario/issues/752), q[uerido-diario#755](https://github.com/okfn-brasil/querido-diario/issues/755), [querido-diario#751](https://github.com/okfn-brasil/querido-diario/issues/751). No quadro? No censo?) 
-- Como contribuir com spiders base? Envia com quantas cidades? Receberemos todas junto? 
-- Issue para cada cidade, ainda que o PR adicione mais de uma? (apenas para o caso de PRs com acréscimo de spider base)

**Comment by ayharano** [comment]
Uso de type hint em código novo

De acordo com https://github.com/scrapy/scrapy/issues/4041, scrapy não fornece no momento stubs para a análise estática via type hint.

Conversei com o @giuliocc e entendo que da estrutura atual de https://github.com/okfn-brasil/querido-diario visa facilitar a contribuição de pessoas tanto em projeto open source como no desenvolvimento de obtenção de diários oficiais municipais com a menor barreira tecnológica possível.

Talvez a questão mais ampla que somente a questão de type hint seria verificar se seria o caso de estabelecer algum conjunto de boas práticas esperadas para o desenvolvimento inicialmente no escopo de spiders.

**Reply by rennerocha** [reply]
Eu não enxergo nenhuma vantagem no uso de type hints no código dos raspadores. Se o objetivo é diminuir a barreira tecnológica, adicionar type hints na minha opinião, vai introduzir uma nova barreira, já que é um assunto que poucos sabem utilizar corretamente e que pode aumentar o ruído no código, dificultando revisões, sem ter nenhum ganho real. A maioria dos raspadores é extremamente simples e não consigo imaginar nada que seria melhorado por adicionarmos type hints.

Na API eu ainda enxergo utilidade, já que usamos o FastAPI que o uso de type hints é fundamental para algumas de suas funcionalidades.

**Reply by ayharano** [reply]
Sobre o meu comentário inicial, depois eu vi que eu queria mencionar coisas antagônicas mesmo mas não inclui explicitamente dessa forma.

A parte que ficou faltando juntando todo o texto seria se deveríamos ter em algum lugar prático o fato, por exemplo, de proibir ou recomendar fortemente que não tenha type hints, dada a questão de diminuir a barreira de entrada.
